components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/dao/IdentityDAO.java

```diff
@@ -182,45 +177,125 @@ private void retryOnTOTOUpdate(List<TOTPSecret> updateTOTPSecretList, Connection
     }
 
     /**
-     * To retrieve the list of data in IDN_IDENTITY_USER_DATA_TEMP as chunks.
+     * To retrieve the list of data in IDN_IDENTITY_USER_DATA_TEMP.
      *
-     * @param startIndex        The start index of the data chunk.
+     * @param syncId            The syncId of the data.
      * @param keyRotationConfig Configuration data needed to perform the task.
      * @return List comprising of the records in the table.
      * @throws KeyRotationException Exception thrown while retrieving data from IDN_IDENTITY_USER_DATA_TEMP.
      */
-    public List<TempTOTPSecret> getTempTOTPSecrets(int startIndex, KeyRotationConfig keyRotationConfig)
+    public List<TempTOTPSecret> getTempTOTPSecrets(int syncId, KeyRotationConfig keyRotationConfig)
             throws KeyRotationException {
 
         List<TempTOTPSecret> totpSecretList = new ArrayList<>();
-        String query = DBConstants.GET_TEMP_TOTP_SECRET;
-        int firstIndex = startIndex;
-        int secIndex = DBConstants.CHUNK_SIZE;
```

Comment:
Don't we consider chunk size now?

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/22/commits/5789ff819be406845886b585a17a8dc5c148b2b0.

Reply:
We haven't used chunk size for Temp table queries. For other tables, the chunk size is being used. For temp tables, I think fetching all the records at once is justified since the number of records can be expected to be low. WDYT?